frontend/src/components/Analysis/BarChart/TableViewModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { Modal, Table } from "antd";
+import React from "react";
+
+const decimalPlaces = 3;
+
+interface Props {
+  title: string;
+  visible: boolean;
+  onClose: () => void;
+  systemNames: string[];
+  xValues: string[];
+  yValues: number[][];
+  xLabel?: string;
+  yLabel?: string;
+  yAxisMax?: number;
+  confidenceScoresList: number[][][];
+  numbersOfSamplesList: number[][];
+}
+
+export function TableViewModal({
+  title,
+  visible,
+  onClose,
+  systemNames,
+  xLabel,
+  yLabel,
+  yAxisMax,
+  xValues,
+  yValues,
+  confidenceScoresList,
+  numbersOfSamplesList,
+}: Props) {
+  yAxisMax = yAxisMax === undefined ? 1 : yAxisMax;
+  const formattedxAxisData = xValues.map((x) => x.replace("\n|\n", "-"));
+  const trimmedConfidenceScores = confidenceScoresList.map(
```

Comment:
The "trimming" happens in different places for `confidenceScoresList` and `yValues`. Could we move them to the same place?